transactions/README.md

```diff
@@ -0,0 +1,63 @@
+# Transactions
+
+### Overview
+
+This is a starter app that is to be used with the Getting Started with Plaid Transactions [YouTube tutorial](https://www.youtube.com/). It implements a very basic personal finance app using /transactions/sync to retrieve and update the user's transaction data. This application uses HTML/VanillaJS on the front end, and NodeJS/Express on the backend, alongside a SQLite database.
+
+### Running the app
+
+If you want the most complete instructions for running the app, please follow along with the video tutorial linked above. The following abbreviated instructions are for those of you who really don't want to watch videos.
+
+#### Clone the repo
+
+Clone the tutorial resources repo to your machine and cd into the project's start directory:
+
+```bash
+git clone https://github.com/plaid/tutorial-resources && cd transactions/start
+```
+
+#### Set up your environment
+
+This app is designed to be used with Node 16 or higher. For information on installing Node, see [How to install Node.js](https://nodejs.dev/learn/how-to-install-nodejs), and consider using [nvm](https://github.com/nvm-sh/nvm) to easily switch between Node versions.
+
+#### Install dependencies
+
+Install the necessary dependencies:
+
+```bash
+npm install
+```
+
+#### Equip the app with credentials
+
+Copy the included **.env.template** to a file called **.env**.
+
+```bash
+cp .env.template .env
+```
+
+Fill out the contents of the **.env** file with the [client ID and secret in your Plaid dashboard](https://dashboard.plaid.com/team/keys). Make sure to pick the appropriate secret value for the environment you're using. (We recommend using sandbox to start.)
+
+#### (Optional) Configure a webhook
+
+This application also shows you how to make use of webhooks, so your application know when to fetch new transactions. If you wish to use this feature, you'll need to have your webhook server, which is running on port 8001, available to Plaid's server.
+
+A common way to do this is to use a tool like ngrok. If you have ngrok installed, run the following command:
+
+```bash
+ngrok http 8001
+```
+
+And ngrok will open a tunnel from the outside world to port 8001 on your machine. Update the WEBHOOK_URL with the domain that ngrok has generated. Your final URL should look something like `https://abde-123-4-567-8.ngrok.io/server/receive_webhook`
+
+See the [Plaid Webhooks Tutorial](https://www.youtube.com/watch?v=0E0KEAVeDyc) for a full description of webhooks and how they work.
+
+#### Start the server
+
+Start the app by running the following command:
+
+```bash
+npm run watch
+```
+
+The server will start running on port 8000 and will update whenever you make a change to the server files. To use the app, navigate to `localhost:8000` in your browser.
```

Comment:
soooo when I tried running this it immediately crashed, I think the SQL server to be pre-populated with a user table:

```
Your userID is undefined
Your error:
Error: SQLITE_ERROR: no such table: users
--> in Database#all('SELECT id, username FROM users', [Function (anonymous)])
    at /Users/ahoffer/tutorial-resources/transactions/start/node_modules/sqlite/build/Database.js:189:21
    at new Promise (<anonymous>)
    at Database.all (/Users/ahoffer/tutorial-resources/transactions/start/node_modules/sqlite/build/Database.js:187:16)
    at Object.getUserList (/Users/ahoffer/tutorial-resources/transactions/start/server/db.js:113:27)
    at /Users/ahoffer/tutorial-resources/transactions/start/server/routes/users.js:34:29
    at Layer.handle [as handle_request] (/Users/ahoffer/tutorial-resources/transactions/start/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/ahoffer/tutorial-resources/transactions/start/node_modules/express/lib/router/route.js:144:13)
    at Route.dispatch (/Users/ahoffer/tutorial-resources/transactions/start/node_modules/express/lib/router/route.js:114:3)
    at Layer.handle [as handle_request] (/Users/ahoffer/tutorial-resources/transactions/start/node_modules/express/lib/router/layer.js:95:5) {
  errno: 1,
  code: 'SQLITE_ERROR',
  __augmented: true
}
```

Reply:
Did it fail at first because you had another process running on port 8000/8001? And then maybe you quit the first process and restart this one? If so, I think what happens is the server starts up, creates the database, but then errors out before it's able to add the tables to the database, so you're left in a weird state where the database is created but the tables aren't.

If that's the case, you can fix this by deleting  appdata.db from the database folder and restarting the app. I'll go ahead and add this to the README, since this also happened to me, too.

Reply:
Update the README to reflect this!